### PR TITLE
Color markup

### PR DESCRIPTION
**Pull request by Scellow**
```c#
            var test = new Label("" +
                                        "[#FF00FF]H[]" +
                                        "[#00FFFF]e[]" +
                                        "[#0000FF]l[]" +
                                        "[#FF0000]l[]" +
                                        "[#3300FF]o[] "
                                        +
                                        "[#FF0033]c[]" +
                                        "[#FF33FF]o[]" +
                                        "[#FF00FF]l[]" +
                                        "[#F408AF]o[]" +
                                        "[#44AAFF]r[]" +
                                        "[#220033]e[]" +
                                        "[#FAF02F]d[] "
                                        +
                                        "[#F4603F]w[]" +
                                        "[#FA801F]o[]" +
                                        "[#F1E08F]r[]" +
                                        "[#F0E0BF]l[]" +
                                        "[#FD05FF]d[]", skin);
```

Will produce:

No regex, no garbage, fast, easy to use :)

### CODE REVIEW

**Comment by Scellow** [comment]
It has one limitation

``[#FF00FF]Hello [#00FF00]colored[] world[]`` won't work

Idea that could be implemented:

- [@drawableName] to include an image

I might try to improve it if my game needs these features (it probably won't be the case, so if anyone want to help, go ahead :p)

File: Nez.Portable/PipelineRuntime/BitmapFonts/BitmapFont.cs [hunk 1]
```diff
@@ -95,40 +95,68 @@ internal BitmapFont( BitmapFontRegion[] regions, int lineHeight )
 
 		public string wrapText( string text, float maxLineWidth )
 		{
-			var words = text.Split( ' ' );
```
Comment (Scellow) [inline Nez.Portable/PipelineRuntime/BitmapFonts/BitmapFont.cs]: this was really bad, it create garbage (string[]), a lot if element is dynamic

**Comment by prime31** [comment]
Is there any reason why MarkupText didn’t work for you? I don’t particularly see the use in adding overhead to the text rendering system via string compares. It is definitely a specialized operation that should take a separate path imo.

**Comment by Scellow** [comment]
MarkupText is nice, but creates too much garbage, and isn't compatible with Nez.UI,  i needed something a bit more inlined/transparent 

I can add a bool in the BitmapFont class ``markupEnabled`` like in libGDX ? 

https://github.com/libgdx/libgdx/wiki/Color-Markup-Language

**Comment by prime31** [comment]
MarkupText shouldn’t be creating garbage for you after it’s compiled. It stores the details after parsing the XML.

A bool guard would work or even just make it a separate method altogether so that UI controls can choose which method to use themselves.

**Comment by prime31** [comment]
Closing old PRs due to non-activity

**Pull request closed without merging**